src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -143,31 +154,27 @@
    pool :- jruby-schemas/pool-queue-type]
   (.pollFirst pool timeout TimeUnit/MILLISECONDS))
 
+(schema/defn borrowed-poison-pill
+  [pool :- jruby-schemas/pool-queue-type
+   instance :- PoisonPill]
+  (.putFirst pool instance)
+  (throw (IllegalStateException.
```

Comment:
Oh, I see...this is really awkward looking as it's own function without the context.  I might just leave it where it was, or add a docstring maybe?

Or maybe it's just me?